gax-grpc/src/test/java/com/google/api/gax/grpc/testing/LocalChannelProvider.java

```diff
@@ -105,6 +127,53 @@ public String getTransportName() {
 
   /** Creates a LocalChannelProvider. */
   public static LocalChannelProvider create(String addressString) {
-    return new LocalChannelProvider(addressString);
+    return new LocalChannelProvider(new LocalAddress(addressString), null);
+  }
+
+  public boolean validateSentHeader(String headerKey, Pattern headerPattern) {
+    Metadata.Key<String> key = Metadata.Key.of(headerKey, Metadata.ASCII_STRING_MARSHALLER);
+
+    if (interceptors.isEmpty()) {
+      return false;
+    }
+    for (LocalHeaderInterceptor interceptor : interceptors) {
+      if (interceptor.getSubmittedHeaders().isEmpty()) {
+        return false;
+      }
+      for (Metadata submittedHeaders : interceptor.getSubmittedHeaders()) {
+        String headerValue = submittedHeaders.get(key);
+        if (headerValue == null || !headerPattern.matcher(headerValue).matches()) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  private static class LocalHeaderInterceptor implements ClientInterceptor {
+    private final ClientInterceptor delegate;
```

Comment:
We use the terminology "innerX" in other places; I'd prefer to be consistent with that pattern (here: "innerInterceptor")

Reply:
Renamed to `innerInterceptor`